src/config/mongoose_config_parser_toml.erl

```diff
@@ -61,15 +61,17 @@ parse_file(FileName) ->
 process(Content) ->
     Config = parse(Content),
     Hosts = get_hosts(Config),
+    HostTypes = get_host_types(Config),
     {FOpts, Config1} = lists:partition(fun(Opt) -> is_function(Opt, 1) end, Config),
     {Overrides, Opts} = lists:partition(fun({override, _}) -> true;
                                            (_) -> false
                                         end, Config1),
-    HOpts = lists:flatmap(fun(F) -> lists:flatmap(F, Hosts) end, FOpts),
-    AllOpts = Opts ++ HOpts,
+    HostsOpts = lists:flatmap(fun(F) -> lists:flatmap(F, Hosts ++ HostTypes) end, FOpts),
+    HostTypesOpts = lists:flatmap(fun(F) -> lists:flatmap(F, HostTypes) end, FOpts),
```

Comment:
This looks redundant to me - it just duplicates some of the options from the previous line. `dedup_state_opts` removes them later.

Reply:
I would rather want to leave it for now as it, it simplifies the following change with modules verifications.

Reply:
I think I was not understood, see my suggestion below.

Reply:
Damn, I think I removed that code at least twice, no idea how it reappeared :)